addons/base_automation/models/base_automation.py

```diff
@@ -1066,9 +1068,12 @@ def _cron_process_time_based_actions(self, *, auto_commit=True):
 
         # retrieve all the automation rules to run based on a timed condition
         automations = self.with_context(active_test=True).search([('trigger', 'in', TIME_TRIGGERS)])
-        for automation_number, automation in enumerate(automations, 1):
-            if auto_commit and not automation.active:
-                # automation deactivated between commits
+        self.env['ir.cron']._commit_progress(remaining=len(automations))
+
+        for automation in automations:
+            automation = automation.exists()
+            if not automation.active:
+                # automation deactivated or disappeared between commits
```

Comment:
I understand the `exists()` but it does one extra sql query for every record. Wouldn't it be possible to instead try to write on the record and ignore the `MissingError`?

Reply:
I tested and we don't have to rollback anything. Most user errors (unless they just wrap a psycopg error) leave the transaction in a good state.